pkg/discovery/discovery.go

```diff
@@ -200,6 +213,23 @@ func Run(restConf *rest.Config, cfg *config.Config) (errCount int) {
 	return errCount
 }
 
+// dumpPlugin will marshal the plugin to the appropriate location in the outputDir:
+// plugins/<name>/definition.json. This makes the data more clear for any consumer
+// looking at the tarball about what was.
+func dumpPlugin(p plugin.Interface, outputDir string) error {
+	b, err := json.Marshal(p)
+	if err != nil {
+		return errors.Wrapf(err, "encoding plugin %v definition to yaml", p.GetName())
+	}
+
+	err = ioutil.WriteFile(
+		filepath.Join(outputDir, results.PluginsDir, p.GetName(), pluginDefinitionFilename),
+		b,
+		os.FileMode(0666),
```

Comment:
Minor point, but would 644 be enough for the permissions here? That's the permissions that other files in the resulting tarball use.

Reply:
👍 Thanks for pointing that out; I just chose a random value and thought I'd come back to it and forgot. That would be fine for me.